File: src/modules/user/validators/user_delete_validator.py
from src.modules.user.models import User
from src.shared.exceptions.not_found_exception import NotFoundException


def user_delete_validator(user_id: int):

    user = User.objects.filter(id=user_id, deleted_at=None).first()

    if user is None:
        raise NotFoundException("Usuário não encontrado!")

File: src/modules/user/validators/user-delete.validator.ts
import { NotFoundException } from '@nestjs/common';
import { PrismaService } from 'src/infra/database/prisma.service';

export async function userDeleteValidator(userId: number) {
  const prisma = new PrismaService();

  const user = await prisma.user.findFirst({
    where: {
      id: userId,
      deletedAt: null,
    },
  });

  if (!user) throw new NotFoundException('Usuário não encontrado!');
}
